src/restruct_poc/solver/constraints.hpp

```diff
@@ -11,47 +11,86 @@
 
 namespace openturbine {
 
+inline int NumConstraintDOFs(const std::vector<Constraint>& constraints) {
+    int num_dofs = 0;
+    for (const auto& constraint : constraints) {
+        num_dofs += constraint.NumDOFs();
+    }
+    return num_dofs;
+}
```

Comment:
This can be done as a `std::transform_reduce`

Reply:
Will update to use transform_reduce